frontend/lib/src/components/widgets/ButtonGroup/ButtonGroup.tsx

```diff
@@ -222,6 +222,27 @@ function getInitialValue(
   return storedValue ?? element.default
 }
 
+function findSegmentsEdges(buttonGroup: Element): void {
+  buttonGroup.querySelectorAll("button").forEach(button => {
+    button.classList.remove("starting-child")
+    button.classList.remove("ending-child")
+    if (
+      button.previousElementSibling === null ||
+      button.offsetLeft <=
+        (button.previousElementSibling as HTMLButtonElement).offsetLeft
+    ) {
+      button.classList.add("starting-child")
+    }
+    if (
+      !button.nextElementSibling ||
+      button.offsetLeft >=
+        (button.nextElementSibling as HTMLButtonElement).offsetLeft
+    ) {
+      button.classList.add("ending-child")
+    }
+  })
+}
```

Comment:
Not sure if this is ideal... But I needed `querySelectorAll` to access DOM elements and use `previousElementSibling` and `nextElementSibling` APIs for the use case here.

If we should avoid this, 
1. perhaps find a way to do this with React `forwardRef` to select the children elements
2. stick to the overflow design Jessi proposed